### PR TITLE
SAAS-9410: Added logs in the installedPackageGeneratedDependencies filter in Salesforce

### PR DESCRIPTION
**Pull request by tamtamirr**
Added logs in the **installedPackageGeneratedDependencies** filter for better visibility.

---

I've added the `getNamespaceSync` to avoid unnecessary async code.

Here's an example for the added logs:

- Namespaces Log
About to add InstalledPackage generated dependencies to Elements from the following namespaces: ["SBQQ","sbaa","standard","ccrz"]

- Summary Log
2023-10-31T13:50:26.006Z debug salesforce-adapter/filters/installed_package_generated_dependencies some Metadata Instances are not under the InstalledPackages directory. summary: {"Layout":{"count":7,"example":"salesforce.Layout.instance.Order_SBQQ__CPQ_Order_Layout@buuss"},"QuickAction":{"count":3,"example":"salesforce.QuickAction.instance.Opportunity_SBQQ__CreateOrderLightning@vuu"},"ValidationRule":{"count":10,"example":"salesforce.ValidationRule.instance.OrderItem_SBQQ__CannotChangeBookingIndicatorOnActivation"},"FieldSet":{"count":11,"example":"salesforce.FieldSet.instance.OpportunityLineItem_sbaa__DealDesk"},"WebLink":{"count":18,"example":"salesforce.WebLink.instance.Order_SBQQ__EditOrderProducts"}}

---
_Release Notes_: 
_None_

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/salesforce-adapter/src/filters/installed_package_generated_dependencies.ts [hunk 1]
```diff
@@ -13,37 +13,54 @@
...
+const { isDefined } = values
 const { awu } = collections.asynciterable
+const { makeArray } = collections.array
+const log = logger(module)
 
 const addInstalledPackageReference = async (
   element: Element,
   installedPackageNamespaceToRef: multiIndex.Index<[string], ElemID>
-): Promise<void> => {
+): Promise<boolean> => {
```
Comment (tamtamirr) [inline packages/salesforce-adapter/src/filters/installed_package_generated_dependencies.ts:40]: This change is in order to determine which elements were actually affected here (The dependency was successfully created)

File: packages/salesforce-adapter/src/filters/installed_package_generated_dependencies.ts [hunk 2]
```diff
@@ -13,37 +13,54 @@
...
 
 const filterCreator: LocalFilterCreator = ({ config }) => ({
   name: 'installedPackageGeneratedDependencies',
   onFetch: async (elements: Element[]) => {
+    const knownInstancesNamespaces = _.uniq(
+      elements
+        .map(getNamespaceSync)
+        .filter(isDefined)
+    )
+    log.debug(`About to add InstalledPackage generated dependencies to Elements from the following namespaces: ${safeJsonStringify(knownInstancesNamespaces)}`)
```
Comment (tomsellek) [inline packages/salesforce-adapter/src/filters/installed_package_generated_dependencies.ts:63]: Nit: aren't we supposed to use format-specifiers-and-params for logs?

File: packages/salesforce-adapter/src/filters/installed_package_generated_dependencies.ts [hunk 3]
```diff
@@ -54,14 +71,21 @@ const filterCreator: LocalFilterCreator = ({ config }) => ({
...
+      .toArray()
     // CustomFields of Standard Objects
-    await awu(elements)
+    const affectedFields = await awu(elements)
       .filter(isObjectType)
       .filter(isStandardObject)
       .flatMap(standardObject => Object.values(standardObject.fields))
-      .forEach(standardObject => addInstalledPackageReference(standardObject, installedPackageNamespaceToRef))
+      .filter(standardObject => addInstalledPackageReference(standardObject, installedPackageNamespaceToRef))
+      .toArray()
```
Comment (tomsellek) [inline packages/salesforce-adapter/src/filters/installed_package_generated_dependencies.ts:83]: Isn't the result of this call an array of booleans?

Comment (tamtamirr) [reply packages/salesforce-adapter/src/filters/installed_package_generated_dependencies.ts:83]: This is filter, not map 🙏

Comment (tomsellek) [reply packages/salesforce-adapter/src/filters/installed_package_generated_dependencies.ts:83]: Oof. How do you say יצאתי חומוס in English?

**Review by tomsellek — APPROVED** [review]

**Pull request merged**